From 8b198a3f47bccaa203ea9ded66952a2b9afef3ad Mon Sep 17 00:00:00 2001
From: Moody Liu <mooodyhunter@outlook.com>
Date: Tue, 5 Sep 2023 00:47:48 +0800
Subject: [PATCH] clike compilers: fix cross_* functions' include

A standard C library may not exist for cross-compile
environments, thus the existence of <stdio.h> cannot be
guaranteed. Use <stddef.h>, which comes directly from
the compiler, instead.
---
 mesonbuild/compilers/mixins/clike.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mesonbuild/compilers/mixins/clike.py b/mesonbuild/compilers/mixins/clike.py
index 4999d6054a63..51943302da41 100644
--- a/mesonbuild/compilers/mixins/clike.py
+++ b/mesonbuild/compilers/mixins/clike.py
@@ -502,7 +502,7 @@ def _compile_int(self, expression: str, prefix: str, env: 'Environment',
                      extra_args: T.Union[None, T.List[str], T.Callable[[CompileCheckMode], T.List[str]]],
                      dependencies: T.Optional[T.List['Dependency']]) -> bool:
         t = f'''{prefix}
-        #include <stdio.h>
+        #include <stddef.h>
         int main(void) {{ static int a[1-2*!({expression})]; a[0]=0; return 0; }}'''
         return self.compiles(t, env, extra_args=extra_args,
                              dependencies=dependencies)[0]
@@ -582,7 +582,7 @@ def cross_sizeof(self, typename: str, prefix: str, env: 'Environment', *,
         if extra_args is None:
             extra_args = []
         t = f'''{prefix}
-        #include <stdio.h>
+        #include <stddef.h>
         int main(void) {{
             {typename} something;
             return 0;
@@ -621,7 +621,7 @@ def cross_alignment(self, typename: str, prefix: str, env: 'Environment', *,
         if extra_args is None:
             extra_args = []
         t = f'''{prefix}
-        #include <stdio.h>
+        #include <stddef.h>
         int main(void) {{
             {typename} something;
             return 0;
